### PR TITLE
Include spatial_ref_sys in upgrade

### PR DESCRIPTION
**Pull request by pramsey**
Change spatial_ref_sys to a multi-value insert, and include it in upgrade scripts.

### CODE REVIEW

**Comment by strk** [comment]
Shall you mention https://trac.osgeo.org/postgis/ticket/5024 in the commit log ?
And all docs would need update, which is why we probably want to decide on a fixed strategy about it.
For example, should we define fixed range rather than having pg_extention contain a 517 lines SQL statement ?

**Comment by strk** [comment]
I'm not sure about what makes this back-portable becuase it's effectively changing behaviour of upgrades for those who intentionally wipe out their spatial_ref_sys after initial creation (to keep DB size low?). Could be very very tiny population.

**Comment by strk** [comment]
I'm thinking for [TRAC-5024](https://trac.osgeo.org/postgis/ticket/5024) 
we might as well just ship an updated spatial_ref_sys.sql without
including it in the postgis_upgrade script, then everything will be
backward compatible and still allow users to get new entries by
loading the newspatial_ref_sys.sql

File: extensions/postgis/Makefile.in [hunk 1]
```diff
@@ -51,7 +51,9 @@ EXTENSION_UPGRADE_SCRIPTS = \
 	sql/raster_unpackage.sql \
 	sql/postgis_upgrade.sql \
 	../../doc/postgis_comments.sql \
-	../postgis_extension_helper_uninstall.sql
+	../postgis_extension_helper_uninstall.sql \
```
Comment (robe2) [inline extensions/postgis/Makefile.in:54]: What if you move this line after the spatial_ref_sys lines to see if it fixes the unpackaged issue.

**Comment by robe2** [comment]
> Well, if the intent is to just address [TRAC-5024](https://trac.osgeo.org/postgis/ticket/5024) in a back-portable way, I'd say this is mostly ready to go (yes, I will squash and reference the ticket in the commit). I guess this is my way of saying that changing and documenting ranges, etc, is a completely different work track. Valid! But not something that has to be addressed here. And this can be ported back, I dunno how far? cc/ @robe2

Yes lets just focus on fixing https://trac.osgeo.org/postgis/ticket/5024 and yes this is ready to go after you do the unpackage fix.

Also since this will make the upgrade scripts much fatter, I'm thinking we should squeeze in the 

SOLUTION 3 discussed in [RFC 7](https://trac.osgeo.org/postgis/wiki/PostGISExtensionPaths)

```
prior-version -- 3.2.x
3.2.x -- 3.2.2
```
as a separate ticket item of course after you commit this.

I will ticket SOLUTION 3 as a back-portable solution. It is introducing different files, but since we always have different files anyway for each release I assume this is legit for backward portable (except maybe for those 3.2.x which might appear as up-gradable steps).  I think that is an okay thing.  If I go from 3.2.0 to 3.2.x, nothing has upgraded and no damage done either unless I then try to then uninstall 3.2.2.  Uninstalling 3.2.2 after upgrading to 3.2.x seems like something no one would even think to do.

I think this needs some more testing (with solution 3) before I feel confident.  So I'm thinking of shipping 3.2.1 without this and having (this feature + RFC 7 Solution 3) in the next micros of 3.1.6, 3.2.2, 3.0.6 

Given we've waited this long, another 3 months for a release is bearable.

@pramsey  @strk thoughts, concerns?

**Comment by pramsey** [comment]
Yes, do not delay our 3.2.1 patch release. And yes, let's do this in small chunks. First this. Then new chaining / zero byte files trick in separate tranche.

File: extensions/postgis/Makefile.in (hunk 1 above)
Comment (pramsey) [reply extensions/postgis/Makefile.in]: I have no idea what that changes, but I'm game for anything!

Comment (pramsey) [reply extensions/postgis/Makefile.in]: Arg, drone is unconveniently busted right now https://cloud.drone.io/postgis/postgis/2710

**Comment by pramsey** [comment]
Green on the CI, @strk, @robe2

**Comment by strk** [comment]
While I'm ok with testing different things on the master branch I'm not sure I like the idea of adding spatial_ref_sys population in the upgrade scripts of micro releases. This is not necessarely a bug we're fixing, there's no documentation saying that upgrading the extension would upgrade the spatial_ref_sys entries. Users might have learnt that they can deal with sptial_ref_sys manually (truncating and loading the spatial_ref_sys.sql file?).

So my position on this is that it's ok to change spatial_ref_sys.sql to use single statement and ON CONFLICT DO NOTHING but it's better not to include in the upgrade procedure for stable branches. For master branch I'm also ok with including in upgrade for now but I do think things can still change before we go final (like: completely dropping spatial_ref_sys.sql and replacing it with a C function could be a good idea)

**Comment by pramsey** [comment]
I don't honestly care either way, I just thought that this was my penance for getting in the way of the view proposal: an alternative that still allowed [spatial_ref_sys updates on upgrade](https://trac.osgeo.org/postgis/ticket/5024). If we don't actually want spatial_ref_sys updates on upgrade...? @robe2

**Comment by strk** [comment]
> I don't honestly care either way, I just thought that this was my penance for getting in the way of the view proposal: an alternative that still allowed [spatial_ref_sys updates on upgrade](https://trac.osgeo.org/postgis/ticket/5024). If we don't actually want spatial_ref_sys updates on upgrade...? @robe2

Better discuss on mailing list about this I guess

**Pull request closed without merging**